### DISCUSSION TITLE
Introduce help entry and add options to smartlog

### DISCUSSION THREAD

**Discussion by john-h-k** [arxanas/git-branchless#729] [Ideas]
I have been using git branchless for only about 3 weeks now but I find it utterly invaluable as a tool. That being said, the lack of man pages for most commands can be a bit of a pain, and particularly the lack of options/manual on how to reduce the volume of data shown by smartlog. I would be happy to help contribute these features if there was a discussion around what would be added (I have peeked at the codebase and have a vague gist of how things work)

**Comment by arxanas** [comment]
Hi @john-h-k, thanks for your interest! The best reference for the smartlog at present would be https://github.com/arxanas/git-branchless/wiki/Command:-git-smartlog. Feel free to add a section/touch-up anything there.

Unfortunately, actually generating and installing man-pages turns out to be difficult (see https://github.com/arxanas/git-branchless/pulls?q=man-pages for my attempt). `cargo` itself does not have any mechanism to install man-pages; this should be provided by the system package manager. However, a plurality of users(/survey respondents) install via `cargo install`, so we wouldn't want to lock documentation behind a system that most users can't access.

<img width="749" alt="Chart of respondents to question How did you install git-branchless?" src="https://user-images.githubusercontent.com/454057/210674431-c9214b1a-6e5d-4ff5-82d8-b6f6d8c1b984.png">

Table of same data:

| Method | Count | Percentage |
| --- | --- | --- |
| `cargo install` | 7 | 43.8% |
| Nix/NixOS | 4 | 25% |
| Package manager | 3 | 18.8% |
| GitHub actions | 1 | 6.3% |
| From source | 1 | 6.3% |

**Reply by john-h-k** [reply]
> The best reference for the smartlog at present would be https://github.com/arxanas/git-branchless/wiki/Command:-git-smartlog. Feel free to add a section/touch-up anything there.

Wonderful, thanks. I'll take a look


About the man pages, I myself don't currently know how the git internal ones work (`git <cmd> --help`), but I would be surprised if there wasn't a good way for extensions to do it. For me, I find that `git branchless smartlog --help` works but `git smartlog --help` doesn't which is frustrating

**Reply by arxanas** [reply]
Git is fairly inflexible on this. When I last checked the source code, I found that you can't really customize the behavior of `--help`. But you can customize the man viewer that's invoked, which could then figure out where to read the man pages from, which I think is what my last attempted workaround does.
